tests/CommunityToolkit.HighPerformance.UnitTests/Extensions/Test_ArrayPoolExtensions.cs

```diff
@@ -87,4 +87,62 @@ public void Test_ArrayExtensions_Clear()
         Assert.AreNotSame(array, backup);
         Assert.IsTrue(backup.AsSpan(0, 16).ToArray().All(i => i == 0));
     }
+
+    [TestMethod]
+    [ExpectedException(typeof(ArgumentOutOfRangeException))]
+    public void Test_ArrayPoolExtensions_EnsureCapacity_InvalidSize()
+    {
+        int[]? array = null;
+
+        ArrayPool<int>.Shared.EnsureCapacity(ref array, -1);
+    }
+
+    [TestMethod]
+    public void Test_ArrayPoolExtensions_EnsureCapacity_NewArray()
```

Comment:
Could you also add a test with the input array being of exactly the same capacity as the new requested capacity?

Reply:
I've added a test for EnsureCapacity after Rent with the same size; not sure if it's worth it to test more precisely as the exact size of the rented array isn't carved in stone. Let me know if you think of something.